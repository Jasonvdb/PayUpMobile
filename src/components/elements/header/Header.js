/**
 * Home screen with bets and transaction summary
 *
 * @format
 * @flow
 */

import React from "react";
import { StyleSheet, View, Text } from "react-native";
import PropTypes from "prop-types";
import * as Animatable from "react-native-animatable";

import ProfileButton from "./ProfileButton";

const Header = props => {
	const {
		children,
		allowFontScaling,
		style,
		onProfilePress,
		subTitle,
		animate,
		...rest
	} = props;

	const TextComponent = animate ? Animatable.Text : Text;

	return (
		<View style={styles.root}>
			<View style={styles.titleContainer}>
				<TextComponent animation="fadeInLeft" style={styles.subTitle}>
					{subTitle || ""}
				</TextComponent>
				<TextComponent animation="fadeInRight" style={styles.title}>
					{children || "PayUp"}
				</TextComponent>
			</View>
			{onProfilePress ? <ProfileButton onPress={onProfilePress}/> : null}
		</View>
	);
};

Header.propTypes = {
	onProfilePress: PropTypes.func,
	children: PropTypes.string,
	subTitle: PropTypes.string,
	animate: PropTypes.bool
};

const styles = StyleSheet.create({
	root: {
		display: "flex",
		flexDirection: "row",
		justifyContent: "space-between",
		alignItems: "center",
		flex: 1,
		paddingLeft: 18,
		paddingRight: 18
	},
	titleContainer: {},
	title: {
		color: "#171D33",
		fontWeight: "600",
		fontSize: 32
	},
	subTitle: {
		color: "#A6AAB4",
		fontSize: 14,
		marginBottom: 4
	}
});
export default Header;
