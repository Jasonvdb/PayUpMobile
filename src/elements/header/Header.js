/**
 * Home screen with bets and transaction summary
 *
 * @format
 * @flow
 */

import React from "react";
import { StyleSheet, View, Text } from "react-native";
import PropTypes from "prop-types";
import ProfileButton from "./ProfileButton";

const Header = props => {
	const { children, allowFontScaling, style, onProfilePress, ...rest } = props;
	return (
		<View style={styles.root}>
			<View style={styles.titleContainer}>
				<Text style={styles.subTitle}>Your balance</Text>
				<Text style={styles.title}>{children || "AppSats"}</Text>
			</View>
			{onProfilePress ? <ProfileButton onPress={onProfilePress}/> : null}
		</View>
	);
};

Header.propTypes = {
	onProfilePress: PropTypes.func
};

const styles = StyleSheet.create({
	root: {
		display: "flex",
		flexDirection: "row",
		justifyContent: "space-between",
		alignItems: "center",
		flex: 1,
		paddingLeft: 18,
		paddingRight: 18
	},
	titleContainer: {},
	title: {
		color: "#171D33",
		fontWeight: "600",
		fontSize: 32
	},
	subTitle: {
		color: "#A6AAB4",
		fontSize: 14,
		marginBottom: 4
	}
});
export default Header;
