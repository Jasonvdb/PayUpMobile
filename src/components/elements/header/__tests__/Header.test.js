/**
 * @format
 */

import "react-native";
import React from "react";
import { shallow, render } from "enzyme";

import Header from "../Header";

describe("Header", () => {
	describe("Rendering", () => {
		const origConsole = console.error;
		beforeEach(() => {
			console.error = message => {
				if (message.indexOf("PascalCase" > -1)) {
					return;
				}

				origConsole(message);
			};
		});
		afterEach(() => {
			console.error = origConsole;
		});

		it("should match to snapshot WITHOUT profile button", () => {
			const props = {};

			props.navigation = { state: { routes: [], index: 0 } };

			const component = render(<Header {...props}/>);
			expect(component).toMatchSnapshot();
		});

		it("should match to snapshot WITH profile button", () => {
			const props = {};

			props.navigation = { state: { routes: [], index: 0 } };

			const component = render(<Header {...props} onProfilePress={() => {}}/>);
			expect(component).toMatchSnapshot();
		});

		it("animated should match to snapshot", () => {
			const props = {};

			props.navigation = { state: { routes: [], index: 0 } };

			const component = render(<Header animate {...props}/>);
			expect(component).toMatchSnapshot();
		});
	});
});
