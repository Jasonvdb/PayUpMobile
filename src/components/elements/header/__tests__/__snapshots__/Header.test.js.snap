// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Rendering animated should match to snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "display:flex;flex-direction:row;justify-content:space-between;align-items:center;flex:1;padding-left:18px;padding-right:18px",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "style": "color:#A6AAB4;font-size:14px;margin-bottom:4px;opacity:0;transform:[object Object]",
            },
            "children": Array [],
            "name": "text",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "style": "color:#171D33;font-weight:600;font-size:32px;opacity:0;transform:[object Object]",
              },
              "children": Array [
                Object {
                  "data": "PayUp",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "text",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "style": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
          Object {
            "attribs": Object {
              "style": "color:#171D33;font-weight:600;font-size:32px;opacity:0;transform:[object Object]",
            },
            "children": Array [
              Object {
                "data": "PayUp",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "text",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "style": "color:#A6AAB4;font-size:14px;margin-bottom:4px;opacity:0;transform:[object Object]",
              },
              "children": Array [],
              "name": "text",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "style": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "view",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Header Rendering should match to snapshot WITH profile button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "display:flex;flex-direction:row;justify-content:space-between;align-items:center;flex:1;padding-left:18px;padding-right:18px",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "animation": "fadeInLeft",
              "style": "color:#A6AAB4;font-size:14px;margin-bottom:4px",
            },
            "children": Array [],
            "name": "text",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "animation": "fadeInRight",
                "style": "color:#171D33;font-weight:600;font-size:32px",
              },
              "children": Array [
                Object {
                  "data": "PayUp",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "text",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "animation": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "animation": undefined,
                "style": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "animation": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "animation": undefined,
              "style": undefined,
            },
          },
          Object {
            "attribs": Object {
              "animation": "fadeInRight",
              "style": "color:#171D33;font-weight:600;font-size:32px",
            },
            "children": Array [
              Object {
                "data": "PayUp",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "text",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "animation": "fadeInLeft",
                "style": "color:#A6AAB4;font-size:14px;margin-bottom:4px",
              },
              "children": Array [],
              "name": "text",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "animation": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "animation": undefined,
                "style": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "animation": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "animation": undefined,
              "style": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "style": "opacity:1",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "style": "shadow-color:black;shadow-offset:[object Object];shadow-opacity:0.25px;shadow-radius:4px;border-width:1px;border-color:#F2F2F2;border-radius:16px;border-style:solid",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "source": "[object Object]",
                    "style": "border-radius:16px;width:30px;height:30px",
                  },
                  "children": Array [],
                  "name": "img",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "source": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "source": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "view",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "style": undefined,
              },
            },
          ],
          "name": "view",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "style": "opacity:1",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "style": "shadow-color:black;shadow-offset:[object Object];shadow-opacity:0.25px;shadow-radius:4px;border-width:1px;border-color:#F2F2F2;border-radius:16px;border-style:solid",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "source": "[object Object]",
                  "style": "border-radius:16px;width:30px;height:30px",
                },
                "children": Array [],
                "name": "img",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "source": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "source": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "view",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "attribs": Object {
                "animation": "fadeInLeft",
                "style": "color:#A6AAB4;font-size:14px;margin-bottom:4px",
              },
              "children": Array [],
              "name": "text",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "animation": "fadeInRight",
                  "style": "color:#171D33;font-weight:600;font-size:32px",
                },
                "children": Array [
                  Object {
                    "data": "PayUp",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "text",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "animation": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "animation": undefined,
                  "style": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "animation": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "animation": undefined,
                "style": undefined,
              },
            },
            Object {
              "attribs": Object {
                "animation": "fadeInRight",
                "style": "color:#171D33;font-weight:600;font-size:32px",
              },
              "children": Array [
                Object {
                  "data": "PayUp",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "text",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "animation": "fadeInLeft",
                  "style": "color:#A6AAB4;font-size:14px;margin-bottom:4px",
                },
                "children": Array [],
                "name": "text",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "animation": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "animation": undefined,
                  "style": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "animation": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "animation": undefined,
                "style": undefined,
              },
            },
          ],
          "name": "view",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
    ],
    "name": "view",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Header Rendering should match to snapshot WITHOUT profile button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "display:flex;flex-direction:row;justify-content:space-between;align-items:center;flex:1;padding-left:18px;padding-right:18px",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "animation": "fadeInLeft",
              "style": "color:#A6AAB4;font-size:14px;margin-bottom:4px",
            },
            "children": Array [],
            "name": "text",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "animation": "fadeInRight",
                "style": "color:#171D33;font-weight:600;font-size:32px",
              },
              "children": Array [
                Object {
                  "data": "PayUp",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "text",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "animation": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "animation": undefined,
                "style": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "animation": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "animation": undefined,
              "style": undefined,
            },
          },
          Object {
            "attribs": Object {
              "animation": "fadeInRight",
              "style": "color:#171D33;font-weight:600;font-size:32px",
            },
            "children": Array [
              Object {
                "data": "PayUp",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "text",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "animation": "fadeInLeft",
                "style": "color:#A6AAB4;font-size:14px;margin-bottom:4px",
              },
              "children": Array [],
              "name": "text",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "animation": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "animation": undefined,
                "style": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "animation": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "animation": undefined,
              "style": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "view",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
